Create updated concept model template (#1207)
ihmeuw/vivarium_research: docs/source/models/templates/concept_model_template.rst

.. _concept_model_template:

.. todo::

>   Replace the above ".. _concept_model_template:" with ".. _{YOUR_MODEL_NAME}:""

Reviewer: I think these lines should remain in place for the template because the `.. _concept_model_template:` target allows other documentation pages to link to the template page specifically. When individual concept model pages are created, they should each get their own label of the form `.. _{YOUR_MODEL_TITLE}:`.

.. _concept_model_template:

.. todo::

>   Replace the above ".. _concept_model_template:" with ".. _{YOUR_MODEL_NAME}:""

Reviewer 2: It seems like we'd want to keep this.

in this section might include the following: 

* Brief background of topic
* Purpose or aims
* Funder information
* Examples of similar analyses

.. _{YOUR_MODEL_SHORT_NAME}2.0:

2.0 Simulation Design
++++++++++++++++++++++

This section is intended as the main body of documentation of your simulation. Information
in this section might include the following: 

* Concept model diagram 
* Links to relevant components of diagram
* Scenario descriptions
* Default specifications 
* Demographics
* Location description
* Effect size

.. _{YOUR_MODEL_SHORT_NAME}2.1:

2.1 Concept model diagram 
-------------------------

.. _{YOUR_MODEL_SHORT_NAME}2.2:

2.2 Default specifications 
--------------------------

The below table is intended to outline the default specifications of your simulation. 
Included in the table is a column of parameter definitions. Please delete this column as you 
fill out the table. 

.. list-table:: Default simulation specifications
  :header-rows: 1

  * - Parameter
    - Definition
    - Value
    - Note
  * - Location(s)
    - Target/simulated region & GBD ID 
    - e.g. Ethiopia (ID: 179)
    -
  * - Number of draws

>     - Desired number of draws that a given simulation is to be run for. (Generally, this should be a number between 0 and 999.)

Reviewer 3: ```suggestion
    - Desired number of draws that a given simulation is to be run for. (Generally, this should be a number between 1 and 999.)
```
I think we need at least one :)

Author: Good point! Lol (-:

Reviewer: Wouldn't the maximum be 1000 then, not 999?

    - e.g. 50 draws 
    - Read more about draws in Vivarium :ref:`here <vivarium_best_practices_monte_carlo_uncertainty>`.
  * - Population size per draw
    - Desired simulated population size per draw for a given simulation. 
    - e.g. 100,000 simulants

>     - Read more about how to determine a reasonable population size for your simulation 
>       :ref:`here <vivarium_best_practices_monte_carlo_uncertainty>`.

Reviewer 3: ```suggestion
    - Read more about how to determine a reasonable population size for your simulation 
      :ref:`here <vivarium_best_practices_monte_carlo_uncertainty>. Note that engineers will
      decide how many random seeds to use to achieve the specified population size per draw.`.
```

  * - Age start (initialization)
    - If sim duration > age start for observation, age_start at initialization will be zero and
      fertility will need to be included. Otherwise, age start at initialization will be age start for 
      observation minus simulation duration and a closed cohort (no fertility).
    - e.g. 0 months
    - Currently vivarium can only have new simulants enter the simulation at age zero, although 
      changing this could be a potential framework improvement. 
  * - Age start (observation)
    - Age at which simulants are included in observer.
    - e.g. 6 months
    -
  * - Age end
    - Age at which simulants are no longer included in observer. 
    - e.g. 5 years
    -
  * - Exit age
    - Age at which simulants age out of the simulation. 
    - e.g. 5 years
    -
  * - Simulation start date
    - 
    - e.g. 2021-07-01
    -
  * - Simulation observation start date
    - 
    - e.g. 2022-01-01
    -
  * - Simulation end date
    - 
    - e.g. 2026-12-31
    -

>   * - Timestep
>     - Amount of time that passes in simulation between each observer.
>     - e.g. 4 days
>     - Read more about how to determine a reasonable timestep for your simulation 
>       :ref:`here <vivarium_best_practices_time_steps>`.

Reviewer 3: ```suggestion
  * - Timestep
    - Amount of time that passes in simulation between each observer.
    - e.g. 4 days
    - Read more about how to determine a reasonable timestep for your simulation 
      :ref:`here <vivarium_best_practices_time_steps>`.
  * - Randomness key columns
    - Parameters used to identify identical simulants across scenarios
    - e.g. ['entrance_time', 'maternal_id', 'sex']
    - Entrance time is currently required as a randomness key column 
  
```

Let's include randomness key columns in this table too!

    - Read more about how to determine a reasonable population size for your simulation 
      :ref:`here <vivarium_best_practices_monte_carlo_uncertainty>`.
  * - Age start (initialization)
    - If sim duration > age start for observation, age_start at initialization will be zero and
      fertility will need to be included. Otherwise, age start at initialization will be age start for 
      observation minus simulation duration and a closed cohort (no fertility).
    - e.g. 0 months
    - Currently vivarium can only have new simulants enter the simulation at age zero, although 
      changing this could be a potential framework improvement. 
  * - Age start (observation)
    - Age at which simulants are included in observer.
    - e.g. 6 months
    -
  * - Age end
    - Age at which simulants are no longer included in observer. 
    - e.g. 5 years
    -
  * - Exit age
    - Age at which simulants age out of the simulation. 
    - e.g. 5 years
    -
  * - Simulation start date
    - 
    - e.g. 2021-07-01
    -
  * - Simulation observation start date
    - 
    - e.g. 2022-01-01
    -
  * - Simulation end date
    - 
    - e.g. 2026-12-31
    -
  * - Timestep

>     - Amount of time that passes in simulation between each observer.

Reviewer 2: Maybe this was copied from somewhere, but "between each observer" doesn't sound quite right to me -- there are other reasons to have a short time step that don't have to do with observation. Maybe:

```suggestion
    - Amount of time that passes in simulation between each "step," which is when events and observations can happen.
```

    - Parameters used to identify identical simulants across scenarios
    - e.g. ['entrance_time', 'maternal_id', 'sex']
    - Entrance time is currently required as a randomness key column 
  

.. _{YOUR_MODEL_SHORT_NAME}2.4:

2.4 Population description
~~~~~~~~~~~~~~~~~~~~~~~~~~

Describe the simulated population demographics here. Information in this section might include: 
  - Cohort type
  - Cohort length
  - Age and sex structure
  - Time step
  - Fertility
  - Stratifications 

.. _{YOUR_MODEL_SHORT_NAME}2.5:

2.5 Location description
~~~~~~~~~~~~~~~~~~~~~~~~

Describe the location (country and/or region(s)) of the simulation here.

.. _{YOUR_MODEL_SHORT_NAME}3.0:

3.0 Verification & validation (V&V) tracking
--------------------------------------------

This section is intended for tracking the progress of V&V of simulation
results. 

The below tables can be filled out iteratively as new model runs are requested and later V&V'd. 
 
.. list-table:: Model runs
  :header-rows: 1

  * - Run number
    - Run description
    - Scenarios
    - Specification modifications
    - Stratification modifications
  * - e.g. 1.0
    - e.g. Baseline concept model updates

>     - e.g. 50 draws; 200,000 pop size

Author: Oh yes good catch -  that is supposed to be for an 'Observer modification' column, which I evidently forgot to add. Adding now!

    - e.g. Count data results stratified by random seed for optimization
    - e.g. Remove children under 6 months from observers

.. note::

  Depending on your simulation and preference, the above table could also be converted to a subheading format
  (i.e., in the event the table gets too lengthy!)

.. list-table:: Model verification and validation tracking
   :widths: 3 10 20
   :header-rows: 1

   * - Run number
     - V&V criteria
     - V&V summary
   * - e.g. 1.0 
     - e.g. Confirm that there is no variation in person-time quantity between different observers of same measure.

>      - e.g. V&V notebooks for model 1.0 can be found here [insert Github link]. V&V criteria satisfied. 

Author: Will add that note!

* Scenario descriptions
* Default specifications 
* Demographics
* Location description
* Effect size

.. _{YOUR_MODEL_SHORT_NAME}2.1:

2.1 Concept model diagram 
-------------------------

.. _{YOUR_MODEL_SHORT_NAME}2.2:

2.2 Default specifications 
--------------------------

The below table is intended to outline the default specifications of your simulation. 
Included in the table is a column of parameter definitions. Please delete this column as you 
fill out the table. 

.. list-table:: Default simulation specifications
  :header-rows: 1

  * - Parameter
    - Definition
    - Value
    - Note
  * - Location(s)
    - Target/simulated region & GBD ID 
    - e.g. Ethiopia (ID: 179)
    -
  * - Number of draws
    - Desired number of draws that a given simulation is to be run for. (Generally, this should be a number between 1 and 1,000.)
    - e.g. 50 draws 
    - Read more about draws in Vivarium :ref:`here <vivarium_best_practices_monte_carlo_uncertainty>`.
  * - Population size per draw
    - Desired simulated population size per draw for a given simulation. 
    - e.g. 100,000 simulants
    - Read more about how to determine a reasonable population size for your simulation 

>       :ref:`here <vivarium_best_practices_monte_carlo_uncertainty>. Note that engineers will
>       decide how many random seeds to use to achieve the specified population size per draw.`.

Reviewer: ```suggestion
      :ref:`here <vivarium_best_practices_monte_carlo_uncertainty>`. Note that engineers will
      decide how many random seeds to use to achieve the specified population size per draw.
```
I think a misplaced backtick got introduced in commit 07c66ad4c59bbad62613d11b9c54879935b2a72f, which is probably what's causing the build to fail.

  * - Age start (initialization)
    - If sim duration > age start for observation, age_start at initialization will be zero and
      fertility will need to be included. Otherwise, age start at initialization will be age start for 
      observation minus simulation duration and a closed cohort (no fertility).
    - e.g. 0 months
    - Currently vivarium can only have new simulants enter the simulation at age zero, although 
      changing this could be a potential framework improvement. 
  * - Age start (observation)

>     - Age at which simulants are included in observer.

Reviewer: Which observer? All of them? Should it be "observers" instead of "observer"?

    - e.g. 6 months
    -
  * - Age end

>     - Age at which simulants are no longer included in observer. 

Reviewer: Same comment: Should it be "observers" instead of "observer"?

    -
  * - Simulation observation start date
    - 
    - e.g. 2022-01-01
    -
  * - Simulation end date
    - 
    - e.g. 2026-12-31
    -
  * - Timestep
    - Amount of time that passes in simulation between each "step," which is when events and observations can happen.
    - e.g. 4 days
    - Read more about how to determine a reasonable timestep for your simulation 
      :ref:`here <vivarium_best_practices_time_steps>`.
  * - Randomness key columns
    - Parameters used to identify identical simulants across scenarios
    - e.g. ['entrance_time', 'maternal_id', 'sex']
    - Entrance time is currently required as a randomness key column 
  

.. _{YOUR_MODEL_SHORT_NAME}2.4:

2.4 Population description
~~~~~~~~~~~~~~~~~~~~~~~~~~

Describe the simulated population demographics here. Information in this section might include: 
  - Cohort type
  - Cohort length
  - Age and sex structure
  - Time step
  - Fertility
  - Stratifications 

.. _{YOUR_MODEL_SHORT_NAME}2.5:

2.5 Location description
~~~~~~~~~~~~~~~~~~~~~~~~

Describe the location (country and/or region(s)) of the simulation here.

.. _{YOUR_MODEL_SHORT_NAME}3.0:

3.0 Verification & validation (V&V) tracking

> --------------------------------------------

Reviewer: Should this section be one level out, i.e., `++++++` like sections 2.0 and 4.0 instead of `------`?
